src/lib-components/lists/DetailList.vue

```diff
@@ -1,45 +1,44 @@
 <script setup lang="ts">
+import { computed, ref, watch } from "vue"
 import { useAppFlasher } from "@/composables"
-import { ref, watch } from "vue"
+import { DetailListConfig, DetailListAPI } from "@/composables/list"
 import BaseAPI from "../../api/base"
 import DateFilter from "../layout/DateFilter.vue"
 import Paginator from "../navigation/Paginator.vue"
 
-const props = withDefaults(
-  defineProps<{
-    refreshTrigger?: number
-    reloadTrigger?: number
-    title: string
-    url: string
-  }>(),
-  {
-    refreshTrigger: 0,
-    reloadTrigger: 0,
-  }
-)
+const props = defineProps<{
+  config: DetailListConfig
+}>()
```

Comment:
I'd suggest we flatten these props.  Tables made use of generics so that we could gain some notion of type safety on the actions and columns configurations.  We don't have a need for that here currently, and instead it just means we'll find ourselves defining configuration props inside the script tag without any additional benefits.  I'd rather just put what will be mostly static configuration props right on the component in the template.

It's notable that `refreshTrigger` and `reloadTrigger` are dropped from support here, as Archive makes use of `reloadTrigger`.  I think that's okay, as `defineExpose` has us covered, and it would be nice to narrow our usage to one way of doing this kind of thing.  @profsmallpine any issues here?  You'd need to expose the `reload` method from `ReviewsList.vue` so your parent can call it.

Reply:
Yeah, that works